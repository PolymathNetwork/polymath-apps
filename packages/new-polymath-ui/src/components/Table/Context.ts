import React from 'react';
import { Api } from 'react-table';

export const Context = React.createContext<
  | Api & {
      tableBodyEl: Element | null;
      tableToolbarEl: Element | null;
      isTableEmpty: boolean;
    }
  | null
>(null);
