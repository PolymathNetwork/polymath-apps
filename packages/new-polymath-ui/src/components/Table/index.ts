import { Row as CsvParserRow, Cell as CsvParserCell } from './useCsvParser';
import { Row as SelectRowRow } from './useSelectRow';

export * from './Table';
export * from './RowActions';

export interface Row extends CsvParserRow, SelectRowRow {}
export interface Cell extends CsvParserCell {}
