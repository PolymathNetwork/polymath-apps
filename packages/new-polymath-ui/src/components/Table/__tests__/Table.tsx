import React from 'react';
import { Table } from '../Table';
import { columns, dataTests } from '../docs/data';
import { render, fireEvent } from '~/testUtils/helpers';

describe('Table', () => {
  test('renders without crashing', () => {
    const { container } = render(
      <Table
        {...{
          data: dataTests,
          columns,
        }}
      >
        <Table.BatchActionsToolbar />
        <Table.Rows small />
        <Table.Pagination />
      </Table>
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  test('clicking select all button selects every rows', () => {
    const rendered = render(
      <Table
        {...{
          data: dataTests,
          columns,
        }}
        selectable
      >
        <Table.BatchActionsToolbar />
        <Table.Rows />
      </Table>
    );

    const { getByText, getAllByTestId, getByTestId } = rendered;

    fireEvent.click(getByTestId('select-all-rows'));

    const selectCount = getByText(/items selected/i);
    const rowsCheckbox = getAllByTestId('select-row');

    expect(
      rowsCheckbox
        .map(checkbox => (checkbox as HTMLInputElement).checked)
        .find(isChecked => !isChecked)
    ).toBeFalsy();

    expect(selectCount).toHaveTextContent(`${dataTests.length} items selected`);
  });

  test('clicking batch actions cancel button hides it', () => {
    const rendered = render(
      <Table
        {...{
          data: dataTests,
          columns,
        }}
        selectable
      >
        <Table.BatchActionsToolbar />
        <Table.Rows />
      </Table>
    );

    const { getByText, getByTestId } = rendered;

    const cancelBatchActionsBtn = getByText(/Cancel/i);

    fireEvent.click(getByTestId('select-all-rows'));

    expect(cancelBatchActionsBtn).toBeVisible();

    fireEvent.click(cancelBatchActionsBtn);

    expect(cancelBatchActionsBtn).not.toBeVisible();
  });
});
