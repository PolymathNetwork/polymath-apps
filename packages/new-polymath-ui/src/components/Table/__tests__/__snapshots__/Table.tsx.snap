// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table renders without crashing 1`] = `
.c8 {
  display: inline-block;
  vertical-align: middle;
  width: 0.7em;
  height: 0.7em;
}

.c8 svg {
  display: block;
  width: 100%;
  height: 100%;
}

.c20 {
  display: inline-block;
  vertical-align: middle;
  width: 10px;
  height: 10px;
}

.c20 svg {
  display: block;
  width: 100%;
  height: 100%;
}

.c24 {
  display: inline-block;
  vertical-align: middle;
  width: 0.9em;
  height: 0.9em;
}

.c24 svg {
  display: block;
  width: 100%;
  height: 100%;
}

.c14 {
  box-sizing: border-box;
  height: 100%;
  text-align: right;
  margin-left: auto;
}

.c19 {
  box-sizing: border-box;
  margin-left: 16px;
}

.c22 {
  box-sizing: border-box;
  margin-right: 16px;
}

.c21 {
  box-sizing: border-box;
  margin-left: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c16 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  border: 2px solid transparent;
  padding: 0.5rem 1rem;
  min-height: 2.5rem;
  line-height: 1.15;
  -webkit-transition-duration: 150ms;
  transition-duration: 150ms;
  -webkit-transition-property: background,color,border-color,opacity;
  transition-property: background,color,border-color,opacity;
  font-family: 'Overpass',sans-serif;
  font-size: 0.875rem;
  font-weight: 500;
  background-color: transparent;
  border-color: transparent;
  color: currentColor;
}

.c16:hover {
  color: #fff;
  background-color: #5596E6;
}

.c16:disabled {
  opacity: 1;
}

.c16button,
.c16input[type='button'],
.c16input[type='submit'],
.c16input[type='reset'],
.c16input[type='file'] {
  border-radius: 0;
}

.c16:disabled:hover {
  background-color: transparent;
  color: currentColor;
}

.c0 {
  box-sizing: border-box;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  font-size: 0.875rem;
  height: 100%;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  height: 100%;
}

.c3 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: solid 1px #ddd;
  position: -webkit-sticky;
  position: sticky;
  top: 0;
  bottom: auto;
  z-index: 1;
  background: #EBF0F7;
  color: #152935;
  border-bottom: none;
}

.c3 + .c2 {
  margin-top: -1px;
}

.c11 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0.6em;
  height: 30px;
}

.c11:first-child {
  padding-left: 23px;
}

.c11:last-child {
  padding-right: 23px;
}

.c4 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0.6em;
  height: 48px;
  font-weight: 500;
}

.c4:first-child {
  padding-left: 23px;
}

.c4:last-child {
  padding-right: 23px;
}

.c6 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  border: 2px solid transparent;
  padding: 0.5rem 1rem;
  min-height: 2.5rem;
  line-height: 1.15;
  -webkit-transition-duration: 150ms;
  transition-duration: 150ms;
  -webkit-transition-property: background,color,border-color,opacity;
  transition-property: background,color,border-color,opacity;
  font-family: 'Overpass',sans-serif;
  font-size: 0.875rem;
  font-weight: 500;
  padding: 0;
  background: none;
  text-transform: none;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  border: none;
  white-space: normal;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  text-align: left;
}

.c6button,
.c6input[type='button'],
.c6input[type='submit'],
.c6input[type='reset'],
.c6input[type='file'] {
  border-radius: 0;
}

.c6 .c7 {
  margin-left: 10px;
}

.c6 .c7 {
  color: #5A6872;
  visibility: hidden;
}

.c18 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: solid 1px #ddd;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  color: #5A6872;
}

.c18 + .c17 {
  margin-top: -1px;
}

.c23 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  border: 2px solid transparent;
  padding: 0.5rem 1rem;
  min-height: 2.5rem;
  line-height: 1.15;
  -webkit-transition-duration: 150ms;
  transition-duration: 150ms;
  -webkit-transition-property: background,color,border-color,opacity;
  transition-property: background,color,border-color,opacity;
  font-family: 'Overpass',sans-serif;
  font-size: 0.875rem;
  font-weight: 500;
  background-color: transparent;
  border-color: transparent;
  color: currentColor;
  color: #3D70B2;
  border-left: solid 1px #ddd;
  border-right: solid 1px #ddd;
  border-top: none;
  border-bottom: none;
  padding-left: 0.8rem;
  padding-right: 0.8rem;
}

.c23:hover {
  color: #fff;
  background-color: #5596E6;
}

.c23:disabled {
  opacity: 1;
}

.c23button,
.c23input[type='button'],
.c23input[type='submit'],
.c23input[type='reset'],
.c23input[type='file'] {
  border-radius: 0;
}

.c23:disabled:hover {
  background-color: transparent;
  color: currentColor;
}

.c23:last-child {
  border-right: none;
}

.c23 .c7 {
  -webkit-transform: rotateZ(0.25turn);
  -ms-transform: rotateZ(0.25turn);
  transform: rotateZ(0.25turn);
}

.c25 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  border: 2px solid transparent;
  padding: 0.5rem 1rem;
  min-height: 2.5rem;
  line-height: 1.15;
  -webkit-transition-duration: 150ms;
  transition-duration: 150ms;
  -webkit-transition-property: background,color,border-color,opacity;
  transition-property: background,color,border-color,opacity;
  font-family: 'Overpass',sans-serif;
  font-size: 0.875rem;
  font-weight: 500;
  background-color: transparent;
  border-color: transparent;
  color: currentColor;
  color: #3D70B2;
  border-left: solid 1px #ddd;
  border-right: solid 1px #ddd;
  border-top: none;
  border-bottom: none;
  padding-left: 0.8rem;
  padding-right: 0.8rem;
}

.c25:hover {
  color: #fff;
  background-color: #5596E6;
}

.c25:disabled {
  opacity: 1;
}

.c25button,
.c25input[type='button'],
.c25input[type='submit'],
.c25input[type='reset'],
.c25input[type='file'] {
  border-radius: 0;
}

.c25:disabled:hover {
  background-color: transparent;
  color: currentColor;
}

.c25:last-child {
  border-right: none;
}

.c25 .c7 {
  -webkit-transform: rotateZ(-0.25turn);
  -ms-transform: rotateZ(-0.25turn);
  transform: rotateZ(-0.25turn);
}

.c15 {
  box-sizing: border-box;
  margin-right: 16px;
  line-height: 1.5;
  font-size: 0.875rem;
  overflow-wrap: break-word;
}

.c13 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: solid 1px #ddd;
  position: absolute;
  top: 0;
  width: 100%;
  background: #252D6B;
  color: white;
  height: 48px;
  border: none;
}

.c13 + .c12 {
  margin-top: -1px;
}

.c13[hidden] {
  visibility: hidden;
  opacity: 0;
}

.c13 .c5 {
  height: 100%;
}

.c10 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: solid 1px #ddd;
}

.c10 + .c9 {
  margin-top: -1px;
}

.c10:hover {
  position: relative;
  background: #e0e8f3;
  border-color: #3D70B2;
}

<div
  class="c0"
>
  <div
    class=""
  />
  <div
    class="c1"
  >
    <div
      data-simplebar="true"
    >
      <div
        class="simplebar-wrapper"
      >
        <div
          class="simplebar-height-auto-observer-wrapper"
        >
          <div
            class="simplebar-height-auto-observer"
          />
        </div>
        <div
          class="simplebar-mask"
        >
          <div
            class="simplebar-offset"
          >
            <div
              class="simplebar-content"
            >
              <div
                class=""
              >
                <div
                  class="c2 c3"
                  style="display: flex; min-width: 680px;"
                >
                  <div
                    class="c4"
                    style="box-sizing: border-box; width: 140px; max-width: 200px;"
                  >
                    <button
                      class="c5 c6"
                      style="cursor: pointer;"
                      tabindex="0"
                      title="Toggle SortBy"
                      type="button"
                    >
                      <span>
                        First Name
                      </span>
                      <span
                        class="c7 c8"
                      >
                        <svg
                          fill-rule="evenodd"
                          height="1em"
                          role="img"
                          viewBox="0 0 10 5"
                          width="1em"
                        >
                          <path
                            d="M0 0l5 4.998L10 0z"
                            fill="currentColor"
                          />
                        </svg>
                      </span>
                    </button>
                  </div>
                  <div
                    class="c4"
                    style="box-sizing: border-box; width: 140px; max-width: 200px;"
                  >
                    <button
                      class="c5 c6"
                      style="cursor: pointer;"
                      tabindex="0"
                      title="Toggle SortBy"
                      type="button"
                    >
                      <span>
                        Last Name
                      </span>
                      <span
                        class="c7 c8"
                      >
                        <svg
                          fill-rule="evenodd"
                          height="1em"
                          role="img"
                          viewBox="0 0 10 5"
                          width="1em"
                        >
                          <path
                            d="M0 0l5 4.998L10 0z"
                            fill="currentColor"
                          />
                        </svg>
                      </span>
                    </button>
                  </div>
                  <div
                    class="c4"
                    style="box-sizing: border-box; width: 100px; max-width: undefinedpx;"
                  >
                    <button
                      class="c5 c6"
                      style="cursor: pointer;"
                      tabindex="0"
                      title="Toggle SortBy"
                      type="button"
                    >
                      <span>
                        Age
                      </span>
                      <span
                        class="c7 c8"
                      >
                        <svg
                          fill-rule="evenodd"
                          height="1em"
                          role="img"
                          viewBox="0 0 10 5"
                          width="1em"
                        >
                          <path
                            d="M0 0l5 4.998L10 0z"
                            fill="currentColor"
                          />
                        </svg>
                      </span>
                    </button>
                  </div>
                  <div
                    class="c4"
                    style="box-sizing: border-box; width: 100px; max-width: undefinedpx;"
                  >
                    <button
                      class="c5 c6"
                      style="cursor: pointer;"
                      tabindex="0"
                      title="Toggle SortBy"
                      type="button"
                    >
                      <span>
                        Visits
                      </span>
                      <span
                        class="c7 c8"
                      >
                        <svg
                          fill-rule="evenodd"
                          height="1em"
                          role="img"
                          viewBox="0 0 10 5"
                          width="1em"
                        >
                          <path
                            d="M0 0l5 4.998L10 0z"
                            fill="currentColor"
                          />
                        </svg>
                      </span>
                    </button>
                  </div>
                  <div
                    class="c4"
                    style="box-sizing: border-box; width: 120px; max-width: undefinedpx;"
                  >
                    <button
                      class="c5 c6"
                      style="cursor: pointer;"
                      tabindex="0"
                      title="Toggle SortBy"
                      type="button"
                    >
                      <span>
                        Status
                      </span>
                      <span
                        class="c7 c8"
                      >
                        <svg
                          fill-rule="evenodd"
                          height="1em"
                          role="img"
                          viewBox="0 0 10 5"
                          width="1em"
                        >
                          <path
                            d="M0 0l5 4.998L10 0z"
                            fill="currentColor"
                          />
                        </svg>
                      </span>
                    </button>
                  </div>
                  <div
                    class="c4"
                    style="box-sizing: border-box; width: 80px; max-width: undefinedpx;"
                  >
                    <button
                      class="c5 c6"
                      style="cursor: pointer;"
                      tabindex="0"
                      title="Toggle SortBy"
                      type="button"
                    >
                      <span />
                      <span
                        class="c7 c8"
                      >
                        <svg
                          fill-rule="evenodd"
                          height="1em"
                          role="img"
                          viewBox="0 0 10 5"
                          width="1em"
                        >
                          <path
                            d="M0 0l5 4.998L10 0z"
                            fill="currentColor"
                          />
                        </svg>
                      </span>
                    </button>
                  </div>
                </div>
                <div
                  class="c9 c10"
                  style="display: flex; min-width: 680px;"
                >
                  <div
                    class="c11"
                    style="display: flex; box-sizing: border-box; width: 140px; max-width: 200px;"
                  >
                    fishing
                  </div>
                  <div
                    class="c11"
                    style="display: flex; box-sizing: border-box; width: 140px; max-width: 200px;"
                  >
                    historian
                  </div>
                  <div
                    class="c11"
                    style="display: flex; box-sizing: border-box; width: 100px; max-width: undefinedpx;"
                  >
                    20
                  </div>
                  <div
                    class="c11"
                    style="display: flex; box-sizing: border-box; width: 100px; max-width: undefinedpx;"
                  >
                    60
                  </div>
                  <div
                    class="c11"
                    style="display: flex; box-sizing: border-box; width: 120px; max-width: undefinedpx;"
                  >
                    <span>
                      <span
                        style="color: rgb(87, 213, 0);"
                      >
                        ●
                      </span>
                       
                      Single
                    </span>
                  </div>
                  <div
                    class="c11"
                    style="display: flex; box-sizing: border-box; width: 80px; max-width: undefinedpx;"
                  />
                </div>
                <div
                  class="c9 c10"
                  style="display: flex; min-width: 680px;"
                >
                  <div
                    class="c11"
                    style="display: flex; box-sizing: border-box; width: 140px; max-width: 200px;"
                  >
                    physics
                  </div>
                  <div
                    class="c11"
                    style="display: flex; box-sizing: border-box; width: 140px; max-width: 200px;"
                  >
                    quality
                  </div>
                  <div
                    class="c11"
                    style="display: flex; box-sizing: border-box; width: 100px; max-width: undefinedpx;"
                  >
                    11
                  </div>
                  <div
                    class="c11"
                    style="display: flex; box-sizing: border-box; width: 100px; max-width: undefinedpx;"
                  >
                    90
                  </div>
                  <div
                    class="c11"
                    style="display: flex; box-sizing: border-box; width: 120px; max-width: undefinedpx;"
                  >
                    <span>
                      <span
                        style="color: rgb(255, 191, 0);"
                      >
                        ●
                      </span>
                       
                      Complicated
                    </span>
                  </div>
                  <div
                    class="c11"
                    style="display: flex; box-sizing: border-box; width: 80px; max-width: undefinedpx;"
                  />
                </div>
                <div
                  class="c9 c10"
                  style="display: flex; min-width: 680px;"
                >
                  <div
                    class="c11"
                    style="display: flex; box-sizing: border-box; width: 140px; max-width: 200px;"
                  >
                    burn
                  </div>
                  <div
                    class="c11"
                    style="display: flex; box-sizing: border-box; width: 140px; max-width: 200px;"
                  >
                    ship
                  </div>
                  <div
                    class="c11"
                    style="display: flex; box-sizing: border-box; width: 100px; max-width: undefinedpx;"
                  >
                    10
                  </div>
                  <div
                    class="c11"
                    style="display: flex; box-sizing: border-box; width: 100px; max-width: undefinedpx;"
                  >
                    52
                  </div>
                  <div
                    class="c11"
                    style="display: flex; box-sizing: border-box; width: 120px; max-width: undefinedpx;"
                  >
                    <span>
                      <span
                        style="color: rgb(255, 191, 0);"
                      >
                        ●
                      </span>
                       
                      Complicated
                    </span>
                  </div>
                  <div
                    class="c11"
                    style="display: flex; box-sizing: border-box; width: 80px; max-width: undefinedpx;"
                  />
                </div>
                <div
                  class="c9 c10"
                  style="display: flex; min-width: 680px;"
                >
                  <div
                    class="c11"
                    style="display: flex; box-sizing: border-box; width: 140px; max-width: 200px;"
                  >
                    wine
                  </div>
                  <div
                    class="c11"
                    style="display: flex; box-sizing: border-box; width: 140px; max-width: 200px;"
                  >
                    reading
                  </div>
                  <div
                    class="c11"
                    style="display: flex; box-sizing: border-box; width: 100px; max-width: undefinedpx;"
                  >
                    18
                  </div>
                  <div
                    class="c11"
                    style="display: flex; box-sizing: border-box; width: 100px; max-width: undefinedpx;"
                  >
                    34
                  </div>
                  <div
                    class="c11"
                    style="display: flex; box-sizing: border-box; width: 120px; max-width: undefinedpx;"
                  >
                    <span>
                      <span
                        style="color: rgb(255, 191, 0);"
                      >
                        ●
                      </span>
                       
                      Complicated
                    </span>
                  </div>
                  <div
                    class="c11"
                    style="display: flex; box-sizing: border-box; width: 80px; max-width: undefinedpx;"
                  />
                </div>
                <div
                  class="c9 c10"
                  style="display: flex; min-width: 680px;"
                >
                  <div
                    class="c11"
                    style="display: flex; box-sizing: border-box; width: 140px; max-width: 200px;"
                  >
                    hospital
                  </div>
                  <div
                    class="c11"
                    style="display: flex; box-sizing: border-box; width: 140px; max-width: 200px;"
                  >
                    low
                  </div>
                  <div
                    class="c11"
                    style="display: flex; box-sizing: border-box; width: 100px; max-width: undefinedpx;"
                  >
                    23
                  </div>
                  <div
                    class="c11"
                    style="display: flex; box-sizing: border-box; width: 100px; max-width: undefinedpx;"
                  >
                    93
                  </div>
                  <div
                    class="c11"
                    style="display: flex; box-sizing: border-box; width: 120px; max-width: undefinedpx;"
                  >
                    <span>
                      <span
                        style="color: rgb(87, 213, 0);"
                      >
                        ●
                      </span>
                       
                      Single
                    </span>
                  </div>
                  <div
                    class="c11"
                    style="display: flex; box-sizing: border-box; width: 80px; max-width: undefinedpx;"
                  />
                </div>
                <div
                  class="c9 c10"
                  style="display: flex; min-width: 680px;"
                >
                  <div
                    class="c11"
                    style="display: flex; box-sizing: border-box; width: 140px; max-width: 200px;"
                  >
                    reaction
                  </div>
                  <div
                    class="c11"
                    style="display: flex; box-sizing: border-box; width: 140px; max-width: 200px;"
                  >
                    category
                  </div>
                  <div
                    class="c11"
                    style="display: flex; box-sizing: border-box; width: 100px; max-width: undefinedpx;"
                  >
                    17
                  </div>
                  <div
                    class="c11"
                    style="display: flex; box-sizing: border-box; width: 100px; max-width: undefinedpx;"
                  >
                    56
                  </div>
                  <div
                    class="c11"
                    style="display: flex; box-sizing: border-box; width: 120px; max-width: undefinedpx;"
                  >
                    <span>
                      <span
                        style="color: rgb(255, 46, 0);"
                      >
                        ●
                      </span>
                       
                      Relationship
                    </span>
                  </div>
                  <div
                    class="c11"
                    style="display: flex; box-sizing: border-box; width: 80px; max-width: undefinedpx;"
                  />
                </div>
                <div
                  class="c9 c10"
                  style="display: flex; min-width: 680px;"
                >
                  <div
                    class="c11"
                    style="display: flex; box-sizing: border-box; width: 140px; max-width: 200px;"
                  >
                    letter
                  </div>
                  <div
                    class="c11"
                    style="display: flex; box-sizing: border-box; width: 140px; max-width: 200px;"
                  >
                    produce
                  </div>
                  <div
                    class="c11"
                    style="display: flex; box-sizing: border-box; width: 100px; max-width: undefinedpx;"
                  >
                    6
                  </div>
                  <div
                    class="c11"
                    style="display: flex; box-sizing: border-box; width: 100px; max-width: undefinedpx;"
                  >
                    12
                  </div>
                  <div
                    class="c11"
                    style="display: flex; box-sizing: border-box; width: 120px; max-width: undefinedpx;"
                  >
                    <span>
                      <span
                        style="color: rgb(255, 46, 0);"
                      >
                        ●
                      </span>
                       
                      Relationship
                    </span>
                  </div>
                  <div
                    class="c11"
                    style="display: flex; box-sizing: border-box; width: 80px; max-width: undefinedpx;"
                  />
                </div>
                <div
                  class="c9 c10"
                  style="display: flex; min-width: 680px;"
                >
                  <div
                    class="c11"
                    style="display: flex; box-sizing: border-box; width: 140px; max-width: 200px;"
                  >
                    skirt
                  </div>
                  <div
                    class="c11"
                    style="display: flex; box-sizing: border-box; width: 140px; max-width: 200px;"
                  >
                    impulse
                  </div>
                  <div
                    class="c11"
                    style="display: flex; box-sizing: border-box; width: 100px; max-width: undefinedpx;"
                  >
                    16
                  </div>
                  <div
                    class="c11"
                    style="display: flex; box-sizing: border-box; width: 100px; max-width: undefinedpx;"
                  >
                    52
                  </div>
                  <div
                    class="c11"
                    style="display: flex; box-sizing: border-box; width: 120px; max-width: undefinedpx;"
                  >
                    <span>
                      <span
                        style="color: rgb(255, 46, 0);"
                      >
                        ●
                      </span>
                       
                      Relationship
                    </span>
                  </div>
                  <div
                    class="c11"
                    style="display: flex; box-sizing: border-box; width: 80px; max-width: undefinedpx;"
                  />
                </div>
                <div
                  class="c9 c10"
                  style="display: flex; min-width: 680px;"
                >
                  <div
                    class="c11"
                    style="display: flex; box-sizing: border-box; width: 140px; max-width: 200px;"
                  >
                    police
                  </div>
                  <div
                    class="c11"
                    style="display: flex; box-sizing: border-box; width: 140px; max-width: 200px;"
                  >
                    desire
                  </div>
                  <div
                    class="c11"
                    style="display: flex; box-sizing: border-box; width: 100px; max-width: undefinedpx;"
                  >
                    5
                  </div>
                  <div
                    class="c11"
                    style="display: flex; box-sizing: border-box; width: 100px; max-width: undefinedpx;"
                  >
                    13
                  </div>
                  <div
                    class="c11"
                    style="display: flex; box-sizing: border-box; width: 120px; max-width: undefinedpx;"
                  >
                    <span>
                      <span
                        style="color: rgb(87, 213, 0);"
                      >
                        ●
                      </span>
                       
                      Single
                    </span>
                  </div>
                  <div
                    class="c11"
                    style="display: flex; box-sizing: border-box; width: 80px; max-width: undefinedpx;"
                  />
                </div>
                <div
                  class="c9 c10"
                  style="display: flex; min-width: 680px;"
                >
                  <div
                    class="c11"
                    style="display: flex; box-sizing: border-box; width: 140px; max-width: 200px;"
                  >
                    delivery
                  </div>
                  <div
                    class="c11"
                    style="display: flex; box-sizing: border-box; width: 140px; max-width: 200px;"
                  >
                    art
                  </div>
                  <div
                    class="c11"
                    style="display: flex; box-sizing: border-box; width: 100px; max-width: undefinedpx;"
                  >
                    21
                  </div>
                  <div
                    class="c11"
                    style="display: flex; box-sizing: border-box; width: 100px; max-width: undefinedpx;"
                  >
                    63
                  </div>
                  <div
                    class="c11"
                    style="display: flex; box-sizing: border-box; width: 120px; max-width: undefinedpx;"
                  >
                    <span>
                      <span
                        style="color: rgb(87, 213, 0);"
                      >
                        ●
                      </span>
                       
                      Single
                    </span>
                  </div>
                  <div
                    class="c11"
                    style="display: flex; box-sizing: border-box; width: 80px; max-width: undefinedpx;"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="simplebar-placeholder"
        />
      </div>
      <div
        class="simplebar-track simplebar-horizontal"
      >
        <div
          class="simplebar-scrollbar"
        />
      </div>
      <div
        class="simplebar-track simplebar-vertical"
      >
        <div
          class="simplebar-scrollbar"
        />
      </div>
    </div>
  </div>
  <div
    class="c12 c13"
    hidden=""
  >
    <div
      class="c14"
      height="100%"
    >
      <span
        class="c15"
        font-size="baseText"
      >
        0
         items selected
      </span>
      <button
        class="c5 c16"
        tabindex="0"
        type="button"
      >
        Cancel
      </button>
    </div>
  </div>
  <div
    class="c17 c18"
  >
    <label
      class="c19"
      for="1"
    >
      Items per page
    </label>
    :
    <div
      class="css-12c1uc0 items-per-page-select"
    >
      <div
        class="css-sq30s7"
      >
        <div
          class="css-6o5gw2"
        >
          <div
            class="css-1crx08z"
          >
            10
          </div>
          <input
            class="css-14uuagi"
            id="1"
            readonly=""
            tabindex="0"
            value=""
          />
        </div>
        <div
          class="css-1wy0on6"
        >
          <div
            aria-hidden="true"
            class="css-767nh9"
          >
            <span
              class="c7 c20"
            >
              <svg
                fill-rule="evenodd"
                height="1em"
                role="img"
                viewBox="0 0 10 5"
                width="1em"
              >
                <path
                  d="M0 0l5 4.998L10 0z"
                  fill="currentColor"
                />
              </svg>
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      class="c21"
    >
      <label
        class="c22"
        for="2"
      >
        Page 
        1
         of 
        5
      </label>
      <button
        aria-label="Previous page"
        class="c5 c23"
        disabled=""
        tabindex="0"
        type="button"
      >
        <span
          class="c7 c24"
        >
          <svg
            height="1em"
            role="img"
            viewBox="0 0 12 8"
            width="1em"
          >
            <path
              d="M10.6.25L6 4.95 1.4.25 0 1.65l6 6.1 6-6.1z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
        </span>
      </button>
      <div
        class="css-12c1uc0"
      >
        <div
          class="css-sq30s7"
        >
          <div
            class="css-6o5gw2"
          >
            <div
              class="css-1crx08z"
            >
              1
            </div>
            <input
              class="css-14uuagi"
              id="2"
              readonly=""
              tabindex="0"
              value=""
            />
          </div>
          <div
            class="css-1wy0on6"
          >
            <div
              aria-hidden="true"
              class="css-767nh9"
            >
              <span
                class="c7 c20"
              >
                <svg
                  fill-rule="evenodd"
                  height="1em"
                  role="img"
                  viewBox="0 0 10 5"
                  width="1em"
                >
                  <path
                    d="M0 0l5 4.998L10 0z"
                    fill="currentColor"
                  />
                </svg>
              </span>
            </div>
          </div>
        </div>
      </div>
      <button
        aria-label="Next page"
        class="c5 c25"
        tabindex="0"
        type="button"
      >
        <span
          class="c7 c24"
        >
          <svg
            height="1em"
            role="img"
            viewBox="0 0 12 8"
            width="1em"
          >
            <path
              d="M10.6.25L6 4.95 1.4.25 0 1.65l6 6.1 6-6.1z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
        </span>
      </button>
    </div>
  </div>
</div>
`;
